src/lava/proc/sdn/models.py

```diff
@@ -0,0 +1,296 @@
+
+class AbstractSigmaDeltaModel(AbstractSigmaModel, AbstractDeltaModel):
+    a_in = None
+    s_out = None
+
+    vth = None
+    sigma = None
+    act = None
+    residue = None
+    error = None
+    bias = None
+    wgt_exp = None
+    state_exp = None
+    cum_error = None
+
+    def __init__(self, proc_params: Dict[str, Any]) -> None:
+        super().__init__(proc_params)
+        self.act_mode = self.proc_params['act_fn']
+
+    def activation_dynamics(self, sigma_data: np.ndarray) -> np.ndarray:
+        """Sigma Delta activation dynamics. Unit and ReLU activations are
+        supported.
+
+        Parameters
+        ----------
+        sigma_data : np.ndarray
+            sigma decoded data
+
+        Returns
+        -------
+        np.ndarray
+            activation output
+
+        Raises
+        ------
+        NotImplementedError
+            if activation mode other than Unit or ReLU is encountered.
+        """
+        if self.act_mode == ACTIVATION_MODE.Unit:
+            act = sigma_data + self.bias
+        elif self.act_mode == ACTIVATION_MODE.ReLU:
+            act = ReLU(sigma_data + self.bias)
+        else:
+            raise NotImplementedError(
+                f'Activation mode {self.act_mode} is not implemented.'
+            )
+        return act
+
+    def dynamics(self, a_in_data: np.ndarray) -> np.ndarray:
+        self.sigma = self.sigma_dynamics(a_in_data)
+        act = self.activation_dynamics(self.sigma)
+        s_out = self.delta_dynamics(act)
+        self.act = act
+
+        return s_out
+
+
+@implements(proc=Sigma, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PySigmaModelFloat(AbstractSigmaModel):
+    """ Floating point implementation of Sigma decoding"""
+    a_in = LavaPyType(PyInPort.VEC_DENSE, float)
+    s_out = LavaPyType(PyOutPort.VEC_DENSE, float)
+
+    sigma: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+
+    def __init__(self, proc_params: Dict[str, Any]) -> None:
+        super().__init__(proc_params)
+
+    def run_spk(self) -> None:
+        a_in_data = self.a_in.recv()
+        self.sigma = self.sigma_dynamics(a_in_data)
+        self.s_out.send(self.sigma)
+
+
+@implements(proc=Sigma, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class PySigmaModelFixed(AbstractSigmaModel):
+    """ Fixed point implementation of Sigma decoding"""
+    a_in = LavaPyType(PyInPort.VEC_DENSE, np.int32, precision=24)
+    s_out = LavaPyType(PyOutPort.VEC_DENSE, np.int32, precision=24)
+
+    sigma: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+
+    def __init__(self, proc_params: Dict[str, Any]) -> None:
+        super().__init__(proc_params)
+
+    def run_spk(self) -> None:
+        a_in_data = self.a_in.recv()
+        self.sigma = self.sigma_dynamics(a_in_data)
+        self.s_out.send(self.sigma)
+
+
+@implements(proc=Delta, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyDeltaModelFloat(AbstractDeltaModel):
```

Comment:
Same as above, duplicate code that could be put into a base class for PyDeltaModelFloat and PyDeltaModelFixed.